walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,159 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.ArrayList;
+import java.util.*; 
+
+
+/*
+for the coding challenge, keep the times for the mandatory, and then create a seperate scedule with possbible times, make it a 
+dictionary so we can keep track of how many people can attend from the optional that time and then school teh times withe the largest number of people
+*/
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request){
+        ArrayList<TimeRange> busyMandatory = new ArrayList<>();
+        ArrayList<TimeRange> busyOptional = new ArrayList<>();
+        ArrayList<TimeRange> availableMandatory = new ArrayList<>();
+        ArrayList<TimeRange> availableOptional = new ArrayList<>();
+        ArrayList<TimeRange> available = new ArrayList<>();
+
+        //Mandatory Attendees
+        // adds all time ranges when attendees are busy
+        for(Event event : events){
+            for(String attendee : request.getAttendees()){
+                //adding time frame only if not in there and one attendee is also an attendee of event
+                if(event.getAttendees().contains(attendee) && !busyMandatory.contains(event.getWhen())){
+                    busyMandatory.add(event.getWhen());
+                    break;
+                }
+            }
+        }
+
+        Collections.sort(busyMandatory, TimeRange.ORDER_BY_START);
+
+        //iterate over the list of busy and if an event is containd in another, remove it.
+        Iterator<TimeRange> iter = busyMandatory.iterator();
+        while(iter.hasNext()){
+            TimeRange current = iter.next();
+            for(int checkIndex=0; checkIndex <busyMandatory.size(); checkIndex++){
```

Comment:
indentation

Reply:
I'm not really sure what part of the indentation you wan t me to fix.

Reply:
Now it looks fine. The indentation looks wired yesterday.